pi/main_pi.js

```diff
@@ -81,26 +91,49 @@ function setWorkspace () {
   sendSettings()
 }
 
+function setProject () {
+  updateTasks(document.getElementById('apitoken').value, document.getElementById('wid').value, document.getElementById('pid').value)
+  sendSettings()
+}
+
+async function updateTasks (apiToken, workspaceId, projectId) {
+  try {
+    await getTasks(apiToken, workspaceId, projectId).then(tasksData => {
+      document.getElementById('tid').innerHTML = '<option value="0"></option>'
+      document.getElementById('taskWrapper').classList.remove('hidden')
+      const selectEl = document.getElementById('tid')
+
+      for (taskNum in tasksData) {
+        const optionEl = document.createElement('option')
+        optionEl.innerText = tasksData[taskNum].name
+        optionEl.value = tasksData[taskNum].id.toString()
+        selectEl.append(optionEl)
+      }
+    })
+  } catch (e) {
+    document.getElementById('taskWrapper').classList.add('hidden')
+  }
+}
+
 async function updateProjects (apiToken, workspaceId) {
   try {
     await getProjects(apiToken, workspaceId).then(projectsData => {
       document.getElementById('pid').innerHTML = '<option value="0"></option>'
-      document.getElementById('workspaceError').classList.add('hiddenError')
```

Comment:
@charles-lebrun Was it intentional to remove this line and similar on 103? My understanding is this is what hides the "Please select a workspace!" message that appears when you first add an API token.

Reply:
Everything else looks legit to me, so once this is resolved I'll give it a quick test locally then can merge. Thanks for making this change!

Reply:
@charles-lebrun @brentonrace. Sorry to bump, but I use this plugin everyday and would be stoked to see this merge accepted. Have you had any movement on this issue?

Reply:
@NickCotter01 I've been caught up in a project so was hoping Charles would push a fix for this comment, but I should have time at end of this week to tidy it up and merge.

Reply:
@NickCotter01 Change is merged and I've posted a new release.